core/data-transformers/page-info-transformer.ts

```diff
@@ -0,0 +1,23 @@
+import { CursorPaginationInfo } from '@/vibes/soul/primitives/cursor-pagination';
+
+interface PageInfo {
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
+  startCursor: string | null;
+  endCursor: string | null;
+}
+
+/**
+ * Utility function to convert GraphQL PageInfo object into CursorPaginationInfo
+ *
+ * @param {PageInfo | undefined} pageInfo - The PageInfo object to transform
+ * @returns {CursorPaginationInfo}
+ */
+export function pageInfoTransformer(pageInfo: PageInfo | undefined): CursorPaginationInfo {
```

Comment:
🍹 Is there a situation where we'd pass `undefined` here? I would've thought it would come as `null` from GraphQL and never undefined

Reply:
@migueloller In the scenario where `blog` node comes back `null`, this would mean that `blog.posts.pageInfo` would be `undefined`.

Perhaps just for ease-of-use, we could do `PageInfo | null | undefined`?